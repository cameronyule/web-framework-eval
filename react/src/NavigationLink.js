import React from 'react';

const NavigationLink = props => (
  <li><a href={props.href}>{props.title}</a></li>
);

NavigationLink.propTypes = {
  href: React.PropTypes.string.isRequired,
  title: React.PropTypes.string.isRequired,
};

export default NavigationLink;
