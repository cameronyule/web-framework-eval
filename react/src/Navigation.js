import React from 'react';

import NavigationLink from './NavigationLink';

export default class Navigation extends React.Component {
  constructor() {
    super();

    this.state = {data: [
      {id: 1, href: "#", title: "Maternity Clothes"},
      {id: 2, href: "#", title: "New Arrivals"},
      {id: 3, href: "#", title: "Dresses"},
      {id: 4, href: "#", title: "Tops"},
      {id: 5, href: "#", title: "Tees"}
    ]};
  }

  render() {
    const navigationNodes = this.state.data.map(function(navitem) {
      return (
        <NavigationLink key={navitem.id} href={navitem.href} title={navitem.title} />
      )
    });

    return (
      <nav>
        <ul>
          {navigationNodes}
        </ul>
      </nav>
    )
  }
}
