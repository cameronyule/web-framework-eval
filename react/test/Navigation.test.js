import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Navigation from 'Navigation';
import NavigationLink from 'NavigationLink';

describe('Navigation', () => {
  let links;

  beforeEach(() => {
    links = [
      {id: 1, href: '#', title: 'Coats & Jackets'},
      {id: 2, href: '#', title: 'Knitwear'},
      {id: 3, href: '#', title: 'Shirts'},
      {id: 4, href: '#', title: 'Tops'},
      {id: 5, href: '#', title: 'Trousers'}
    ];
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Navigation data={links} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('requires an array of navigation link data', () => {
    expect(() => {
      /* eslint-disable no-console */
      console.error = jest.fn();
      shallow(<Navigation />);
      expect(console.error).toBeCalled();
      expect(console.error.mock.calls[0][0]).toContain('data');
      /* eslint-enable no-console */
    }).toThrow();
  });

  it('creates a NavigationLink per item', () => {
    const wrapper = shallow(<Navigation data={links} />);
    expect(wrapper.find(NavigationLink).length).toEqual(5);
  });

  it('uses the order defined in the input data array', () => {
    const wrapper = shallow(<Navigation data={links} />);
    const linksWrapper = wrapper.find(NavigationLink);

    const ids = links.map((o) => { return o.id; });
    const linkIDs = linksWrapper.map((l) => { return parseInt(l.key()); })

    expect(linkIDs).toEqual(ids);
  });
});
