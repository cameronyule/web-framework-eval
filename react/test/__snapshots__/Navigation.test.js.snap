exports[`Navigation renders correctly 1`] = `
<nav>
  <ul>
    <NavigationLink
      href="#"
      id={1}
      title="Coats & Jackets" />
    <NavigationLink
      href="#"
      id={2}
      title="Knitwear" />
    <NavigationLink
      href="#"
      id={3}
      title="Shirts" />
    <NavigationLink
      href="#"
      id={4}
      title="Tops" />
    <NavigationLink
      href="#"
      id={5}
      title="Trousers" />
  </ul>
</nav>
`;
